bin-prot/tests/layouts.rs

```diff
@@ -162,6 +189,13 @@ fn smoke_test_deserialize_block() {
     let rule = Layout::deserialize(deserializer).unwrap().bin_prot_rule;
 
     let mut de = Deserializer::from_reader_with_layout(BLOCK_BYTES, rule);
-    let _result: Result<Value> = Deserialize::deserialize(&mut de);
-    // This is failing at the moment so don't check the result
+    let block: Value = Deserialize::deserialize(&mut de).expect("Failed to deserialize block");
+
+    assert_eq!(
+        block["t"]["protocol_state"]["t"]["t"]["previous_state_hash"]["t"],
+        Value::String(vec![
+            30, 76, 197, 215, 115, 43, 42, 245, 198, 30, 253, 134, 49, 117, 82, 71, 182, 181, 180,
+            95, 18, 250, 46, 1, 25, 3, 78, 193, 57, 152, 116, 49,
+        ])
+    );
```

Comment:
do we want to check for the other fields in the block as well in this test? also, is there a way to get around the `["t"]`s so we can directly access `block["protocol_state"]["previous_state_hash"]`, or not really due to how the deserializer works?

Reply:
I was thinking I'll write something that flattens the values by squashing out the `{version, t}` parts. In a future PR!